src/java/org/apache/cassandra/cql3/PasswordObfuscator.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3;
+
+/**
+ * Obfuscates passwords in a given string
+ */
+public class PasswordObfuscator
+{
+    public static final String OBFUSCATION_TOKEN = "*******";
+    private static final String PASSWORD_TOKEN = "password";
+
+    public String obfuscate(String sourceString)
```

Comment:
I guess it is not the final version because the existing tests should fail. The issues I can think of:
1. the last part of the query (that is located after the password) is rejected, but the password is not always the last statement
2. the query may have multiple password statements (actual for `BATCH`), there should be a loop
3. password value is quoted, I do not see any logic to calculate indexes of `'`

Reply:
hi @alex-ninja , the way to do it "properly" - covering all the cases possible, it not so simple at all, it gets complicated pretty quickly and there is still yet some other corner case not covered. This whole feature has to be treated differently by parsing CQL via ANTLR (ideally) and just replacing password tokens there - that is the direction we would like to see ideally drift towards, but due to lack of time to implement that and due to refactoring related so close before the release, we just come with "simple and stupid" solution which would obfuscate just everything after `password`, even though it would render the whole CQL statement invalid. The fact that the passwords are not leaked - and we have to be 100% sure of that, is more important than having these clauses obfuscated nicely. We can deal with that later.

Reply:
ok, that makes sense to me!